src/components/DonationForm.jsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import {
+  FormLabel,
+  Input,
+  InputGroup,
+  Select,
+  FormControl,
+  FormErrorMessage,
+  Button,
+} from '@chakra-ui/react';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import './DonationForm.css';
+
+const schema = yup.object({
+  firstName: yup.string().required('Invalid first name'),
+  lastName: yup.string().required('Invalid first name'),
+  zipcode: yup.number().typeError('ZIP code must be a number'),
+  email1: yup.string().email('Invalid email').required('Email required'),
+  email2: yup.string().email('Invalid email').required('Email required'),
+});
+
+function DonationForm() {
+  const {
+    register,
+    // control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+  return (
+    <div className="form-padding">
+      <form onSubmit={handleSubmit(data => console.log(data))}>
+        <div className="field-section">
+          <h1 className="title">Name</h1>
+          <div className="form">
+            <FormControl isInvalid={errors && errors.firstName} width="47%">
```

Comment:
(not a comment to change anything) This line and a bunch of other `FormControl`/`FormErrorMessage` lines were using optional chaining (ie `error.?firstName`) in order to ensure attributes existed, but it's typically better practice to implement it as I did here, with `errors && errors.firstName`. Just something to make a note of for the future